File: RedSheet.DbEntities/Models/OurTeamMember.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("OurTeamMembers", Schema = "dbo")]
	[Serializable]
	public partial class OurTeamMember  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int OurTeamMemberId { get; set; } 
 
	[RelationshipTableAttribue("ProjectNegotionalities","dbo")]
	//Relationships
	public int ProjectNegotionalityId { get; set; } 
 
	[ForeignKey("ProjectNegotionalityId")]
	public ProjectNegotionality ProjectNegotionality { get; set; } 
 
	[RelationshipTableAttribue("TeamRoles","dbo")]
	//Relationships
	public int TeamRoleId { get; set; } 
 
	[ForeignKey("TeamRoleId")]
	public TeamRole TeamRole { get; set; } 
 
	[RelationshipTableAttribue("Users","dbo")]
	//Relationships
	public int UserId { get; set; } 
 
	[ForeignKey("UserId")]
	public User User { get; set; } 
	[InverseProperty("OurTeamMember")]
	public ICollection<OurTeamMemberRequire> OurTeamMemberRequires { get; set; } 

	[InverseProperty("OurTeamMember")]
	public ICollection<OurTeamMemberBehaviour> OurTeamMemberBehaviours { get; set; } 


	public OurTeamMember()
	{
	this.OurTeamMemberRequires = new HashSet<OurTeamMemberRequire>();

	this.OurTeamMemberBehaviours = new HashSet<OurTeamMemberBehaviour>();

    }

    }
}

File: redsheet-portal/src/app/database-models/our-team-member.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectNegotionality, TeamRole, User, OurTeamMemberRequire, OurTeamMemberBehaviour,  vOurTeamMemberRecord  } from './'
export class OurTeamMember {
    constructor(ourTeamMember?: OurTeamMember  | vOurTeamMemberRecord )  {
        let properties = [ "ourTeamMemberId", "projectNegotionalityId", "teamRoleId", "userId", "ourTeamMemberRequires", "ourTeamMemberBehaviours",];
        for (let property of properties)
            if (ourTeamMember && ourTeamMember[property])
                this[property] = ourTeamMember[property];
    }
 
	ourTeamMemberId : number =   0 ;
 
    @range(0,2147483647)
	projectNegotionalityId : number =   undefined;
	projectNegotionality : ProjectNegotionality  ;
 
    @range(0,2147483647)
	teamRoleId : number =   undefined;
	teamRole : TeamRole  ;
 
    @range(0,2147483647)
	userId : number =   undefined;
	user : User  ;
	@nested(OurTeamMemberRequire)
	ourTeamMemberRequires: OurTeamMemberRequire[];

	@nested(OurTeamMemberBehaviour)
	ourTeamMemberBehaviours: OurTeamMemberBehaviour[];



}
